cmd/finch/version.go

```diff
@@ -4,27 +4,108 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"os"
 
 	"github.com/spf13/cobra"
 
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/lima"
 	"github.com/runfinch/finch/pkg/version"
 )
 
-var finchVersion = version.Version
-
-func newVersionCommand() *cobra.Command {
+func newVersionCommand(limaCmdCreator command.LimaCmdCreator, logger flog.Logger) *cobra.Command {
 	versionCommand := &cobra.Command{
 		Use:   "version",
 		Args:  cobra.NoArgs,
 		Short: "Shows Finch version information",
-		RunE:  versionAction,
+		RunE:  newVersionAction(limaCmdCreator, logger).runAdapter,
 	}
 
 	return versionCommand
 }
 
-func versionAction(cmd *cobra.Command, args []string) error {
-	fmt.Println("Finch version:", finchVersion)
+type versionAction struct {
+	creator command.LimaCmdCreator
+	logger  flog.Logger
+}
+
+// NerdctlVersionOutput captures the nerdctl version.
+type NerdctlVersionOutput struct {
+	Client NerdctlClientOuput  `json:"Client"`
+	Server NerdctlServerOutput `json:"Server"`
+}
+
+// NerdctlClientOuput captures the nerdctl Client output.
+type NerdctlClientOuput struct {
+	Version    string                    `json:"Version"`
+	GitCommit  string                    `json:"GitCommit"`
+	GoVersion  string                    `json:"GoVersion"`
+	Os         string                    `json:"Os"`
+	Arch       string                    `json:"Arch"`
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlServerOutput captures the nerdctl Server output.
+type NerdctlServerOutput struct {
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlComponentsOutput captures the nerdctl components output.
+type NerdctlComponentsOutput struct {
+	Name    string `json:"Name"`
+	Version string `json:"Version"`
+	Details struct {
+		GitCommit string `json:"GitCommit"`
+	}
+}
+
+func newVersionAction(creator command.LimaCmdCreator, logger flog.Logger) *versionAction {
+	return &versionAction{creator: creator, logger: logger}
+}
+
+func (va *versionAction) runAdapter(cmd *cobra.Command, args []string) error {
+	return va.printVersion()
+}
+
+func (va *versionAction) printVersion() error {
+	_, err := lima.GetVMStatus(va.creator, va.logger, limaInstanceName)
+	if err != nil {
+		fmt.Println("Finch version:", version.Version)
+	}
+
+	limaArgs := []string{"shell", limaInstanceName, "nerdctl", "version", "--format", "json"}
+	out, _ := va.creator.CreateWithoutStdio(limaArgs...).Output()
```

Comment:
It's better to do the error handling here:
```suggestion
	out, err := va.creator.CreateWithoutStdio(limaArgs...).Output()
	if err != nil {...}
```